src/ucp/api/ucp.h

```diff
@@ -2401,6 +2540,95 @@ ucs_status_ptr_t ucp_am_send_nb(ucp_ep_h ep, uint16_t id,
+ * @note This operation does not support CUDA memory.
+ * @note This operation supports specific flags, which can be passed
+ *       in @a param by @ref ucp_request_param_t.flags. The exact set of flags
+ *       is defined by @ref ucp_send_am_flags.
+ *
+ * @param [in]  ep            UCP endpoint where the Active Message will be run.
+ * @param [in]  id            Active Message id. Specifies which registered
+ *                            callback to run.
+ * @param [in]  header        Active message header.
+ * @param [in]  header_length Active message header length in bytes.
+ * @param [in]  buffer        Pointer to the data to be sent to the target node
+ *                            of the Active Message.
+ * @param [in]  count         Number of elements to send.
+ * @param [in]  param         Operation parameters, see @ref ucp_request_param_t.
+ *
+ * @note Sending only header without actual data is allowed and is recommended
+ *       for transfering latency critical amount of data.
+ * @note The maximum allowed header size can be obtained by querying worker
+ *       attributes by @ref ucp_worker_query routine.
+ *
+ *
+ * @return NULL                 - Active Message was sent immediately.
+ * @return UCS_PTR_IS_ERR(_ptr) - Error sending Active Message.
+ * @return otherwise            - Operation was scheduled for send and can be
+ *                                completed in any point in time. The request
+ *                                handle is returned to the application in order
+ *                                to track progress of the message. If user
+ *                                request was not provided in @a param->request,
+ *                                the application is responsible for releasing
+ *                                the handle using @ref ucp_request_free routine.
+ */
+ucs_status_ptr_t ucp_am_send_nbx(ucp_ep_h ep, uint16_t id,
+                                 const void *header, size_t header_length,
+                                 const void *buffer, size_t count,
+                                 ucp_request_param_t *param);
+
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Receive Active Message sent with rendezvous protocol.
+ *
+ * This routine receives a message that is described by the data descriptor
+ * @a data_desc, local address @a buffer, size @a count and @a param
+ * parameters on the @a worker. The routine is a non-blocking and therefore
+ * returns immediately. The receive operation is considered completed when the
+ * message is delivered to the @a buffer. If the receive operation cannot be
+ * started the routine returns an error.
+ *
+ * @note After this call UCP takes ownership of @a data_desc descriptor, so
+ *       there is no need to release it even if the operation fails.
+ *       The routine returns a request handle instead, which can further be used
+ *       for tracking operation progress.
+ *
+ * @note This operation does not support CUDA memory.
```

Comment:
`Current implementation of the function does not support CUDA memory. ` I'm not sure that this the right place to document this constrain.

Reply:
ok, can remove it (moreover we don't have such notes for others)

Reply:
Probably we should add "known issues" section to news file to list this limitation.

Reply:
I would phrase it as "Currently the active message API supports only host memory buffers"

Reply:
my only concern, that once it is fixed we will forget to update ucp.h